components/openweather_api/actions/get-weather-forecast-by-location/get-weather-forecast-by-location.mjs

```diff
@@ -0,0 +1,105 @@
+import { ConfigurationError } from "@pipedream/platform";
+import openweatherApi from "../../openweather_api.app.mjs";
+
+export default {
+  key: "openweather_api-get-weather-forecast-by-location",
+  name: "Get current weather forcast by location",
+  description: `Retrieves 1-16 days weather forecast for a specified location. 
+  [See the docs here](https://openweathermap.org/forecast16#geo16). For more accurate reading, you are adviced to fill in the country code and/orstate code `,
+  version: "0.0.1",
+  type: "action",
+  props: {
+    openweatherApi,
+    appid: {
+      propDefinition: [
+        openweatherApi,
+        "appId",
+      ],
+    },
+    city: {
+      type: "string",
+      label: "City",
+      description: "City location. For example \"Houston\" in the United States.",
+    },
+    stateCode: {
+      type: "string",
+      label: "State code",
+      description: "State code. For example \"TX\" for Texas.",
+      optional: true,
+    },
+    countryCode: {
+      type: "string",
+      label: "Country Code",
+      description: "The 2 or 3-letter country code, for example \"US\" for the United States.",
+      optional: true,
+    },
```

Comment:
Since `city`, `stateCode`, and `countryCode` are used in both `get-current-weather-by-location` and `get-weather-forecast-by-location`, they should be declared in the app file.
```suggestion
    city: {
      propDefinition: [
        openweatherApi,
        "city",
      ],
    },
    stateCode: {
      propDefinition: [
        openweatherApi,
        "stateCode",
      ],
    },
    countryCode: {
      propDefinition: [
        openweatherApi,
        "countryCode",
      ],
    },
```